src/legacy/ui/public/vis/request_handlers/courier.js

```diff
@@ -101,6 +102,11 @@ const CourierRequestHandlerProvider = function () {
         request.stats(getRequestInspectorStats(requestSearchSource));
 
         try {
+          // Abort any in-progress requests before fetching again
+          if (abortSignal) {
+            abortSignal.addEventListener('abort', () => requestSearchSource.cancelQueued());
```

Comment:
this might not handle postFlightRequests

Reply:
we probably need to pass abortController to https://github.com/elastic/kibana/pull/42035/files#diff-a670a4336113cea11c1f626375311df9R137